lib/matplotlib/axes/_axes.py

```diff
@@ -4952,6 +4973,28 @@ def reduce_C_function(C: array) -> float
                 raise ValueError(
                     "y contains non-positive values, so cannot be log-scaled")
             ty = np.log10(ty)
+
+        if xscale == 'symlog' or yscale == 'symlog':
+            symlog_transform = {
+                'x': mscale.SymmetricalLogTransform(
+                    xscale_kwargs.get('base', 10),
+                    xscale_kwargs.get('linthresh', 2),
+                    xscale_kwargs.get('linscale', 1),
+                ),
+                'y': mscale.SymmetricalLogTransform(
+                    yscale_kwargs.get('base', 10),
+                    yscale_kwargs.get('linthresh', 2),
+                    yscale_kwargs.get('linscale', 1),
+                )
```

Comment:
Instead of using `get` with a default, you should call `xscale_kwargs.setdefault` (after making a copy) and then pass `**xscale_kwargs`. This way any invalid parameters passed as kwargs will fail with an error instead of silently ignored.

Reply:
I'm not sure I understand what you mean here.
With `xscale_kwargs.setdefault`, default values will be set, so I can't see how this could help raising "an error instead of silently ignored".

Reply:
Because we don't want to modify the input dictionary from the user, only a local copy we have.